asset/_lib/chunked-file-reader/index.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const chunkFormatter = require('./formatters');
+
+function _averageRecordSize(array) {
+    return Math.floor(array.reduce((accum, str) => accum + str.length, 0) / array.length);
+}
+
+// [{offset, length}] of chunks `size` assuming `delimiter` for a file with `total` size.
+function getOffsets(size, total, delimiter) {
+    if (total === 0) {
+        return [];
+    }
+    if (total < size) {
+        return [{ length: total, offset: 0 }];
+    }
+    const fullChunks = Math.floor(total / size);
+    const delta = delimiter.length;
+    const length = size + delta;
+    const chunks = [];
+    for (let chunk = 1; chunk < fullChunks; chunk += 1) {
+        chunks.push({ length, offset: (chunk * size) - delta });
+    }
+    // First chunk doesn't need +/- delta.
+    chunks.unshift({ offset: 0, length: size });
+    // When last chunk is not full chunk size.
+    const lastChunk = total % size;
+    if (lastChunk > 0) {
+        chunks.push({ offset: (fullChunks * size) - delta, length: lastChunk + delta });
+    }
+    return chunks;
+}
+
+// This function will grab the chunk of data specified by the slice plus an
+// extra margin if the slice does not end with the delimiter.
+function getChunk(readerClient, slice, opConfig, logger, metadata) {
+    const delimiter = opConfig.line_delimiter;
+
+    async function getMargin(offset, length) {
+        let margin = '';
+        while (margin.indexOf(delimiter) === -1) {
+            // reader clients must return false-y when nothing more to read.
+            const chunk = await readerClient(offset, length); // eslint-disable-line no-await-in-loop, max-len
```

Comment:
We've disabled this eslint rule:

```suggestion
            const chunk = await readerClient(offset, length);
```

Reply:
fixed